### ISSUE TITLE
UMI dedup

### ISSUE THREAD

**Issue by lkerr34** [CGATOxford/UMI-tools#544]
Hi and thanks you for such a useful tool!

I have used umi_tools extract to remove UMIs from some data and this seemed to have been successful. I then use Trimgalore to trim reads, bismark to align reads and samtools to sort and index the alignment. 

However, when I use umi_tools dedup I am only left with only 89 of the original 36986 read pairs. I seen in the documentation that when UMIs are extracted using umi_tools extract, then the UMI is the last word in the read name. I have attached snippets of the output from umi_tools extract and my aligned file. In the output from umi tools the UMI is indeed at the end of the file name but in the alignment file the UMI is the 8 digits before the "_1:N:0..." in the read name so perhaps this is the problem.  I have also attached the output I get from umi_tools dedup. Any help you could provide regarding this issue would be really appreciated!

Thanks!
Lyndsay

![image](https://user-images.githubusercontent.com/55538405/177828372-43af2e44-6737-40e8-adfc-7d501eb3b563.png)
![image](https://user-images.githubusercontent.com/55538405/177828696-e3d8a2ce-c539-4008-97db-a63a6f54c2ed.png)
![image](https://user-images.githubusercontent.com/55538405/177827470-fb567fc3-59e1-4458-b79c-e29a4005a92f.png)

**Comment by IanSudbery** [comment]
Yes. It appears that bismark is altering the readnames so that it is attaching the `1:N:0` part of the read header to the rest of the read name, and thus UMI-tools can't find the UMI. My feeling is that it would be best to remove this. Probably the easiest way is before alignment. I think:

```
$ zcat input_reads.fastq.gz | sed -E 's/ [12]:N:0:.+//' | gzip > reprocessed.fastq.gz
```

should do the trick, but it would be worth checking the output.

**Comment by lkerr34** [comment]
Great--thank you so much for your help!
